### PR TITLE
added VPA SLO alert

### PR DESCRIPTION
**Pull request by QuantumEnigmaa**
towards https://github.com/giantswarm/giantswarm/issues/24111

This PR:

- adds/changes/removes etc

### Checklist

- [X] Update changelog in CHANGELOG.md.
- [ ] Add Unit tests
- [X] Request review from oncall area, as well as team (e.g: `oncall-kaas-cloud` GitHub group).
- [X] Alerting rules must have a comment documenting why it needs to exist.
- [ ] Consider creating a dashboard (if it does not exist already) to help oncallers monitor the status of the issue.

### CODE REVIEW

**Review by marieroque — APPROVED** [review]

File: helm/prometheus-rules/templates/recording-rules/service-level.rules.yml [hunk 1]
```diff
@@ -349,6 +349,30 @@ spec:
...
+    # -- VPA
+    # Amount of request for VPA
+    - expr: "count(up{app=~'vertical-pod-autoscaler.*'}) by (cluster_type,cluster_id)"
+      labels:
+        class: MEDIUM
+        area: platform
+        service: vertical-pod-autoscaler
+      record: raw_slo_requests
+
+    # Amout of errors for VPA
```
Comment (whites11) [inline helm/prometheus-rules/templates/recording-rules/service-level.rules.yml:361]:
```suggestion
    # Amount of errors for VPA
```

File: helm/prometheus-rules/templates/recording-rules/service-level.rules.yml [hunk 2]
```diff
@@ -349,6 +349,30 @@ spec:
...
+    - expr: "sum((up{app=~'vertical-pod-autoscaler.*'} * -1) + 1) by (cluster_id, cluster_type)"
```
Comment (whites11) [inline helm/prometheus-rules/templates/recording-rules/service-level.rules.yml:362]: this is not a trivial query (due to the `* -1 +1` part) and it's not self explanatory. Can you please add a comment to explain how it works?

**Review by mcbenjemaa — APPROVED** [review]

File: helm/prometheus-rules/templates/recording-rules/service-level.rules.yml [hunk 3]
```diff
@@ -349,6 +349,35 @@ spec:
...
+    - expr: "count(up{app=~'vertical-pod-autoscaler.*'}) by (cluster_type,cluster_id)"
+      labels:
+        class: MEDIUM
+        area: platform
+        service: vertical-pod-autoscaler
+      record: raw_slo_requests
+
+    # Amount of errors for VPA
+    # Up metric is set to 1 for each succesful scrape and set to 0 otherwise.
+    # If up made a succesful scrape, there is no error. Up returns 1, multiplied by -1
```
Comment (QuentinBisson) [inline helm/prometheus-rules/templates/recording-rules/service-level.rules.yml:363]:
```suggestion
    # If up made a successful scrape, there is no error. Up returns 1, multiplied by -1
```

**Review by QuentinBisson — APPROVED** [review]
LGTM if tested. Do you mind creating a follow-up PR to explain how to create an SLO alert?

**Review by QuentinBisson — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -168,3 +168,42 @@ Show success
 
 [unit testing rules]: https://prometheus.io/docs/prometheus/latest/configuration/unit_testing_rules/
 
+### SLO Framework integration
+
+In order to incorporate the SLO Framework in the Prometheus rules, several rules need to be implemented :
+* One which will record the amount of requests for the designated target
+* One recording the amount of errors for the same target
+* One recording the targeted availibility (for exemple 99.9% availibilty)
```
Comment (QuentinBisson) [inline README.md:176]:
```suggestion
* One recording the targeted availibility (for exemple 99.9% availabilty)
```

**Review by QuentinBisson — APPROVED** [review]
<3

**Comment by hervenicol** [comment]
No unit tests? :cry:

**Pull request merged**